toutoumomoma.go

```diff
@@ -338,32 +337,41 @@ func streamEntropy(r io.Reader) (entropy, variance float64, err error) {
 		}
 	}
 
-	entropy, variance = entropyVariance(&n, N)
+	entropy, variance = entropyVariance(&counts, n)
 	return entropy, variance, nil
 }
 
-// entropyVariance returns the entropy and entropy variance for counts in n
-// for a sequence that is N long. See https://arxiv.org/pdf/1807.02603.pdf
+// entropyVariance returns the entropy and entropy variance for counts in
+// counts for a sequence that is n long. See https://arxiv.org/pdf/1807.02603.pdf
 // for details of the variance calculation.
-func entropyVariance(n *[256]float64, N int) (entropy, variance float64) {
-	// TODO(kortschak): Use a more numerically stable algorithm for calculating variance.
-
-	if N == 0 {
+func entropyVariance(counts *[256]float64, n int) (entropy, variance float64) {
+	if n == 0 {
 		return 0, 0
 	}
-	// e = -∑i=1..k((p_i)*log(p_i))
-	for _, cnt := range n {
+
+	// H = -∑i=1..k((p_i)*log(p_i))
+	// F² = ∑i=1..k((p_i)*log²(p_i)) - H²
+	//
+	// Variance in H is F²/N
+	//
+	// Calculated using the weighted incremental algorithm for
+	// mean and variance estimates.
```

Comment:
I would add a link here to the Wikipedia page.

Reply:
Done